### PR TITLE
Workaround conda-build issues to fix osx-64 build

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Review by jakirkham — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ source:
   sha256: 98e91ccead4d4756ae3c9cde5e09191a8e586d9f4d50838e7ec09d6411dfdb63
 
 build:
-  number: 0
+  number: 1
 
 requirements:
   build:
```
Comment (h-vetinari) [inline recipe/meta.yaml:15]: There's one more section like this here that could be similarly adapted for consistency:
```
- {{ compiler('c') }}  # [not osx][
    ](https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)# Break circular dependency openssl -> clang -> ld64 -> sigtool -> openssl
    - clang_bootstrap_{{ target_platform }} {{ c_compiler_version }}.*  # [osx]
```

**Comment by jakirkham** [comment]
With that addressed, does this otherwise look good to you, @h-vetinari ? 🙂

**Review by h-vetinari — APPROVED** [review]
> With that addressed, does this otherwise look good to you, @h-vetinari ? 🙂

Absolutely! Thanks a lot for fixing this!

**Comment by h-vetinari** [comment]
Looks like this does what it does (on osx)
```
*** Installing runtime programs
install apps/openssl -> /Users/runner/miniforge3/conda-bld/openssl_split_1648106784813/[...]/bin/openssl
install tools/c_rehash -> /Users/runner/miniforge3/conda-bld/openssl_split_1648106784813/[...]/bin/c_rehash
```

**Comment by jakirkham** [comment]
Yeah it was installing the binaries [before]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=480072&view=logs&j=58ac6aab-c4bc-5de2-2894-98e408cc8ec9&t=933f325c-924e-533d-4d95-e93b5843ce8b&l=6355 ), which is what was observed here ( https://github.com/conda-forge/openssl-feedstock/issues/87#issuecomment-1077301073 ). The issue is conda-build didn't include them in the package because the overlapped with the compiler package, which was also being installed in `host` (instead of `build`) as conda-build was merging these two prefixes. Isuru's fix here ensures they don't get merged so the compiler lives in a separate `build` environment.

**Pull request merged**